conf_collectd_plugin.py

```diff
@@ -0,0 +1,375 @@
+            "enable collectd to collect data from apache?")
+    else:
+        print "You have previously installed this plugin."
+        res = utils.ask("Would you like to reinstall this plugin", no)
+
+    if not res:
+        return
+
+    apache_title()
+    print ""
+    sys.stdout.write(
+        "To enable collectd plugin with Apache, the following "
+        "steps need to be taken:\n"
+        "1. mod_status for apache needs to be enabled. (Default is enabled)\n"
+        "2. ExtendedStatus needs to be turned on.  (Default is off)\n"
+        "3. Enable the server-status handler for each virtual host.\n")
+
+    noop = raw_input("Press Enter to continue")
+    utils.print_step("Begin collectd Apache plugin installer")
+
+    """ check point
+    - check dependency
+    - change system file
+    - check mod status
+    - TODO: pull template
+    - prompt for user information
+    """
+    if not utils.command_exists("curl"):
+        utils.exit_with_failure("Curl is needed for this plugin.")
+
+    # ubuntu check
+    # Assumption:
+    # -latest apache2 is installed and the installation
+    # -directory is in the default place
+    utils.print_step("  Checking if mod_status is enabled")
+    cmd_res = utils.get_command_output("ls /etc/apache2/mods-enabled")
+    if "status.conf" not in cmd_res or "status.load" not in cmd_res:
+        utils.print_step("Enabling apache2 mod_status module.")
+        ret = utils.call_command("sudo a2enmod status")
+        if ret != 0:
+            utils.exit_with_message("a2enmod command was not found")
+    utils.print_success()
+    print ""
+    sys.stdout.write(
+        'In order to enable the apache plugin with collectd, the '
+        'ExtendedStatus setting must be turned on.\n'
+        'This setting can be turned on by having "ExtendedStatus on" '
+        'in one of the .conf file.\n')
+
+    sys.stdout.write(
+        'If you have already enabled this status, '
+        'answer "no" to the next question '
+        'and ignore the following warning.\n'
+        'If you would like us to enable this status, answer "yes" and we will '
+        'include a extendedstatus.conf file in your apache folder.\n')
+
+    res = utils.ask(
+        "Would you like us to enable "
+        "the ExtendedStatus?")
+
+    """
+    missing the flow where user wants to turn on the setting themselves.
+    """
+    if res:
+        # include the config file in /apache2/conf-enabled
+        conf_dir = "/etc/apache2/conf-enabled"
+        utils.print_step("Checking if "+conf_dir+" exists.")
+        if utils.check_path_exists(conf_dir):
+            # pull config file here
+            utils.print_success()
+            include_apache_es_conf(conf_dir)
+            sys.stdout.write(
+                "\nextendedstatus.conf is now included in the " +
+                conf_dir+" dir.\n")
+            print "Restarting apache"
+            ret = utils.call_command(
+                "service apache2 restart >> " +
```

Comment:
oh, and we should tell people we are going to restart apache.  they should know of course, but just in case